### ISSUE TITLE
error: fork/exec /bin/sh: argument list too long

### ISSUE THREAD

**Issue by glennpratt** [pulumi/pulumi-command#285]
### What happened?

Running `pulumi up` on an existing stack on Linux fails.

Poking at the code, I see this is because of PULUMI_COMMAND_STDOUT/PULUMI_COMMAND_STDERR and the previous run containing rather long output.

https://github.com/pulumi/pulumi-command/blob/7c5c4bd68e66950c042c5ce1ad3eb6f9bfda835c/provider/pkg/provider/local/commandOutputs.go#L73-L78

Since I don't use this feature and I don't always have the option of adjusting limits, I'd like an option to turn it off. Furthermore, I'd prefer if the option could be made off by default as well, as I don't generally need this information in state files.

### Example

strace
```
sudo -E strace -E PATH=$PATH -u $USER -f -e trace=execve -s 1000 -v --signals=SIGPWR pulumi up 2> log
```

From log above, we can tell our command is not particularly long, the environment is massive.
```
execve("/bin/sh", ["/bin/sh", "-c", "ansible REDACTED"], ["SHELL=/bin/bash", ...REDACTED..., "PULUMI_COMMAND_STDOUT=ansible-playbook [core 2.14.6]\n  config file REDACTED"...]) = -1 E2BIG (Argument list too long)
```

### Output of `pulumi about`

```
CLI
Version      3.75.0
Go Version   go1.20.5
Go Compiler  gc

Host
OS       ubuntu
Version  20.04
Arch     x86_64

Backend
Name           REDACTED
URL            file://~
User           REDACTED
Organizations
```

### Contributing

Vote on this issue by adding a 👍 reaction. 
To contribute a fix for this issue, leave a comment (and link to your pull request, if you've opened one already).

**Comment by AMKamel** [comment]
Same here:
![image](https://github.com/pulumi/pulumi-command/assets/14953512/05da5943-acde-4b15-9fe2-a748f55f9a1e)
```
CLI          
Version      3.91.1
Go Version   go1.21.3
Go Compiler  gc

Plugins
NAME    VERSION
nodejs  unknown

Host     
OS       ubuntu
Version  22.04
Arch     x86_64
```

**Comment by AMKamel** [comment]
I was only able to work around this by `pulumi stack export` then edit the stack manually to make command `"stdout": ""` then `pulumi stack import`.

This is the reason, also I am heading towards always using `no_log:true` in ansible playbook to avoid this, but what was the objective of storing this stdout/stderr in state?

**Comment by theneva** [comment]
@iwahbe I've just encountered this issue in `@pulumi/command` version `0.11.1` with a `remote.Command`. After some digging, it seems to me like #355 only addresses `local.Command`s. Am I completely missing something?

If I'm not missing anything, is this issue the right place to ask nicely for opt-out support for `remote.Command` as well?

**Comment by julsemaan** [comment]
I originally authored the fix and you're right, this only deals with local.command as this is what I used and had the ability to test easily. I have 0 clue how to use the remote functionality but if you provide a minimal example to use it, I can give a shot at creating a PR with the fix

**Comment by iwahbe** [comment]
CC @thomas11

**Comment by thomas11** [comment]
Thank you @julsemaan, a PR would be awesome! At first glance it seems the code for `remote` works the same way. [Here](https://github.com/pulumi/pulumi-command/blob/4d94d6d1f397ab98ea260ff7495656933fd3b401/provider/pkg/provider/remote/commandOutputs.go#L52) is the equivalent place to where you added `if AddPreviousOutputInEnv ...` for `local`.

Possibly we can share some code between local and remote but we can look at that later.

**Comment by julsemaan** [comment]
Opened a PR: https://github.com/pulumi/pulumi-command/pull/451

It's lacking a test though as I couldn't mock things appropriately for remote.Command

**Comment by MLOpsGuy** [comment]
This issue was closed but I don't understand your answers. I used `local.Command` and I have the `argument list too long` error on update and delete states. Someone can help me, please?

**Comment by thomas11** [comment]
Hi @MLOpsGuy, the previous behavior that can lead to this error is still the default. You need to set `addPreviousOutputInEnv = false` on your `local.Command`  resource.

**Comment by MLOpsGuy** [comment]
Thanks @thomas11, I really appreciate your help
